modules/alloydb/main.tf

```diff
@@ -434,14 +440,12 @@ resource "google_alloydb_instance" "secondary" {
   }
 }
 
-# Read pool (instance_type = "READ_POOL") cannot be created for secondary cluster
-# and does not support the following attributes:
+# Read pool (instance_type = "READ_POOL") does not support the following attributes:
 # * availability_type: Because 1 node pool (read_pool_config.node_count) is always zonal, two or more is always regional.
 # * gce_zone
 # * network_config.enable_outbound_public_ip
-resource "google_alloydb_instance" "read_pool" {
-  provider       = google-beta
-  for_each       = local.read_pool
+resource "google_alloydb_instance" "read_pool_primary" {
```

Comment:
this rename will also cause a recreate. A moved block could help here (but only if you don't change providers)

Reply:
Reverted back the provider.